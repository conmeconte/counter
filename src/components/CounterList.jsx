import React, { Component } from 'react';
import Counter from './Counter';

const styles = {
  header: {
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
  },
  title: {
    textAlign: 'center',
    backgroundColor: 'lightgray',
    margin: 0,
    padding: 20,
    color: 'black',
  },
  btnContainer: {
    display: 'flex'
  },
  addBtn: {
    height: 50,
    backgroundColor: 'green',
    color: 'white',
    width: '50%'
  },
  deleteBtn: {
    height: 50,
    backgroundColor: 'red',
    color: 'white',
    width: '50%'
  },
  counterContainer: {
    padding: 0,
    margin: 0,
  },
  counter: {
    margin: 0,
    padding: 0,
    display: 'flex',
    justifyContent: 'space-between',
    backgroundColor: 'orange'
  }
}

class CounterList extends Component {
  state = {
    counterNum: 1
  }

  addCounter = () => this.setState({ counterNum: this.state.counterNum + 1 });
  removeCounter = () => this.setState({ counterNum: this.state.counterNum - 1 });

  createList = () => {
    let result = [];
    for(let i = 0; i < this.state.counterNum; i++) {
      result.push(<Counter />);
    }

    return result;
  }

  render() {
    return (
      <div style={styles.header}>
        <h1 style={styles.title}>COUNTER LIST</h1>
        
        <div style={styles.btnContainer}>
          <button 
            id='add-counter' 
            onClick={this.addCounter}
            style={styles.addBtn}
          >
            ADD
          </button>

          <button 
            id='remove-counter' 
            onClick={this.removeCounter}
            style={styles.deleteBtn}
          >
            DELETE
          </button>
        </div>

        <ul style={styles.counterContainer}>
          {
            this.createList().map(
              (counter, idx) => <div key={idx} style={styles.counter}>{counter}</div>
            )
          }
        </ul>
      </div>
    )
  }
}

export default CounterList;