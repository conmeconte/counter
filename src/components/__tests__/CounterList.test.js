import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import CounterList from '../CounterList';

describe('CounterList component layout', () => {
  it('matches the snapshot', () => {
    const tree = renderer.create(<CounterList />).toJSON();
    expect(tree).toMatchSnapshot();
  })
  
  it('has add counter button', () => {
    const wrapper = shallow(<CounterList />);
    expect(wrapper.find('#add-counter').exists()).toEqual(true);
  })

  it('has remove counter button', () => {
    const wrapper = shallow(<CounterList />);
    expect(wrapper.find('#remove-counter').exists()).toEqual(true);
  })
  
  it('should render one counter by default', () => {
    const wrapper = shallow(<CounterList />);
    const counters = wrapper.find('Counter');
    
    expect(counters.length).toEqual(1);
  })
})

describe('CounterList component UI', () => {
  it('adds counter when add counter clicked', () => {
    const wrapper = shallow(<CounterList />);
    const button = wrapper.find('#add-counter');
    button.simulate('click');
    const counters = wrapper.find('Counter');

    expect(counters.length).toEqual(2);
  })

  it('removes counter when remove counter clicked', () => {
    const wrapper = shallow(<CounterList />);
    const button = wrapper.find('#remove-counter');
    button.simulate('click');
    const counters = wrapper.find('Counter');

    expect(counters.length).toEqual(0);
  })
})