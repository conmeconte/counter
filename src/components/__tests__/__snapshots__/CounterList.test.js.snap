// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterList component layout matches the snapshot 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }
>
  <h1
    style={
      Object {
        "backgroundColor": "lightgray",
        "color": "black",
        "margin": 0,
        "padding": 20,
        "textAlign": "center",
      }
    }
  >
    COUNTER LIST
  </h1>
  <div
    style={
      Object {
        "display": "flex",
      }
    }
  >
    <button
      id="add-counter"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "green",
          "color": "white",
          "height": 50,
          "width": "50%",
        }
      }
    >
      ADD
    </button>
    <button
      id="remove-counter"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "red",
          "color": "white",
          "height": 50,
          "width": "50%",
        }
      }
    >
      DELETE
    </button>
  </div>
  <ul
    style={
      Object {
        "margin": 0,
        "padding": 0,
      }
    }
  >
    <div
      style={
        Object {
          "backgroundColor": "orange",
          "display": "flex",
          "justifyContent": "space-between",
          "margin": 0,
          "padding": 0,
        }
      }
    >
      <p
        id="count"
        style={
          Object {
            "fontSize": 20,
            "fontWeight": "bold",
            "margin": "auto",
          }
        }
      >
        CLICK COUNT: 
        0
      </p>
      <button
        id="increment-btn"
        onClick={[Function]}
        style={
          Object {
            "padding": 20,
          }
        }
      >
        INCREMENT
      </button>
      <button
        id="decrement-btn"
        onClick={[Function]}
        style={
          Object {
            "padding": 20,
          }
        }
      >
        DECREMENT
      </button>
    </div>
  </ul>
</div>
`;
